app/Rest/RestExample.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Boilerplate\Theme\Rest;
+
+use JustCoded\WP\Framework\Objects\RestController;
+
+/**
+ * Custom json rest endpoint to illustrate like it work
+ */
+class RestExample extends RestController {
+
+	/**
+	 * @var string
+	 */
+	protected $namespace = 'rest_example';
+
+	/**
+	 * @var array
+	 */
+	protected $slugs = [
+		'id'   => '/id/(?P<id>\d+)',
+		'slug' => '/slug/(?P<slug>\S+)',
+	];
+
+	/**
+	 * Init
+	 *
+	 * @throws \Exception
+	 */
+	public function init() {
+		$this->add_route( $this->slugs['id'], 'GET', [ $this, 'example_id' ], [
+			'id' => [
+				'validate_callback' => [ $this, 'validate_id' ]
+			]
+		] );
+
+		$this->add_route( $this->slugs['slug'], 'GET', [ $this, 'example_slug' ], [
+			'slug' => [
+				'validate_callback' => [ $this, 'validate_slug' ]
+			]
+		] );
```

Comment:
routes:

- GET, '', 'list', [$this, 'get_items']
- POST, '', 'create', [$this, 'create_item']
- GET, ':id', 'view', [$this, 'get_item'],
- PUT, ':id', 'update', [$this, 'update_item']
- DELETE, ':id', 'delete', [$this, 'delete_item']